src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseTestCaseResource.java

```diff
@@ -37,29 +42,41 @@
 
     private final ProgrammingExerciseService programmingExerciseService;
 
+    private final AuthorizationCheckService authCheckService;
+
+    private final UserService userService;
+
     public ProgrammingExerciseTestCaseResource(ProgrammingExerciseTestCaseRepository programmingExerciseTestCaseRepository,
-            ProgrammingExerciseTestCaseService programmingExerciseTestCaseService, ProgrammingExerciseService programmingExerciseService) {
+            ProgrammingExerciseTestCaseService programmingExerciseTestCaseService, ProgrammingExerciseService programmingExerciseService,
+            AuthorizationCheckService authCheckService, UserService userService) {
         this.programmingExerciseTestCaseRepository = programmingExerciseTestCaseRepository;
         this.programmingExerciseTestCaseService = programmingExerciseTestCaseService;
         this.programmingExerciseService = programmingExerciseService;
+        this.authCheckService = authCheckService;
+        this.userService = userService;
     }
 
     @GetMapping(value = "programming-exercise/{exerciseId}/test-cases")
```

Comment:
This is not a change from this PR and should be addressed in a different PR as it could break functionality not related to this task. If we want to streamline the URLs, I think it would be better to create a follow up task imo.

Reply:
Yes, you are right about not renaming it in this PR. But add short JavaDoc here please

Reply:
I added the JavaDoc.